### ISSUE TITLE
error line number in quoted lambda eval

### ISSUE THREAD

**Issue by ramagottfried** [HfMT-ZM4/libo#6]
the new line number system is cool, but how should the line numbers be handled when the function call is in a lambda?

at the moment it is giving us the line in the lambda which is awesome, but still incomplete somehow

for example:

<pre><code>
----------begin_max5_patcher----------
392.3ocuR0zSCCCC8b6uhPNMjJaqqUCFRbkybmNgxRcgLRSJooiBS7emDmVF
eI3BhKIxO+d1IO68wQzM5dnkRNmbMIJZebTDB4AhFhin0rdtj0hzn5oPeiYJ
WWBdhIAJVn2homUolb7HZkVYqXbvmY9.lnLTlMaOIajmpqVnjfE6vh2ItU7L
JNcwz4G3p6rijyFPaXV9cB0s2X.tM7cVluzIhrHK2ekep+LyUGx5AMgxXepA
BBnW1IkWw32CVZxODQV60+Rbr+H4Oy1HWPdnSagIRV8lR1jqWmTnHjYb8Nsn
LckKeA8NPJ0jG0FY4QEz.gJs1ka1FloPc7aduTn.ttSg0OezlLPKnrLqPqt4
aY7KirE+Girz7TbxktxecZ1e8LCqFZPeZyGeRd7ONHa0cF9XCGbAxg2TIzZE
JzSeGmr.muzVVSyNvzNPFagaSYq13COKACEpPXJFZfchQ9n4QYFmyYc1VmAe
Vz9k4zfT29kQ0IP0w98SWKwsPEqFZaFlr3xZ7KwuBDZdA7D
-----------end_max5_patcher-----------
</code></pre>

**Comment by ramagottfried** [comment]
also sometimes the error line number is just at the end of the function (I think due to the unbound sub bundle missing error issue).

(I just lost well over an hour trying to figure out where a problem was, I knew it was an unbound address somewhere, but it turned out it was just one extra letter should have been scalar./pitch2pix instead of scalars./pitch2pix, the error messages just told me the last line of the lambda function)

**Comment by ramagottfried** [comment]
for me this is a particularly important error situation to figure out, since I'm using tons of functions, the error messages usually just point to the call of the function, and give little to no information about where in the quoted lambda the problem is. I basically have to debug blind, and try to slowly isolate the problem.

maybe it could be possible to give the function name and the line number in the function?

and if we print the expression that is causing the problem this would be helpful also.

**Comment by maccallum** [comment]
errors that occur inside of a lambda that was compiled during evaluation should now be correct 4a3c7e714f08471799d2a9f18a39bac54b7664ce

**Comment by ramagottfried** [comment]
that's great, any chance to somehow tell which lambda function the line number refers to?

for example, how do you know which function is failing here?  (imagining a more complex situation)

```
<pre><code>
----------begin_max5_patcher----------
419.3ocuSErTBCCD8b6WQLmfYp.skgo3Ld0ydGXbRSSgfgjRZJVkg+cS1zpn
333AGtj18su8ks6a6wv.btpkUiQ2gVfBBNFFD.PNfft3.7NRKUPpAZX0HVak
dDUUvbDi7TLrVCjdboLYvvnkR6KCF1mVvkLppQBbR5.4EdAy2daZOwRkzTye
i4xDmLZRGrrYGWJXl5yq2BpZL8nocnUDCcCWt9IMiZ7eXoYNkPIYYtGyhcmS
mMZBZUWMdYLuVw7EfenQHdjPelYvQ+RzGB3Z6RBEJehC5TXn6H5eZnZX0Fz8
nI14JLYsuuuQYXCDjc4EjAKVYyfPiopCJdQ7ba9k3MLgPgdQoEE2rD6ITpT1
biyI5kxgC6kK4Oom0.WuAAm8xkmu+L49Q2NN9R6N4ZX2ySASOdt6Q7rrqgeC
Dfu+u8OEzlN7utDTqZzz9lnaxf9rOKr1NWRLbk7LNodNWbsjppCLccGY3Jra
YaUZWXVDDxk9PvTvZ1AdOeXfhIZ6zzXGkMZnsvsylh8kZ2M0xFNTcna21dkv
FrjriUW0MMfE8vSguGvga7I
-----------end_max5_patcher-----------
</code></pre>
```

**Comment by maccallum** [comment]
yeah, so that's a nasty case... the line numbers are correct though. in that specific case, we could have apply also report an error that has its own line number

**Comment by ramagottfried** [comment]
hmm yeah, I think that might be worth doing, since the line number by itself doesn’t help you know where to fix the problem

**Comment by maccallum** [comment]
we're going to have to look more carefully at this. what i'm doing now is offsetting the line number of a lambda by the line number of the function that compiled it. but in your example, neither the line number in the lambda (with 1 being the first line of the lambda), nor the line number offset by the location in the file when it was compiled help. and the original line number is lost by the time you get to the second o.expr.codebox. so yeah, i guess having both line numbers will help you piece things together, but it gets complicated very quickly

**Comment by ramagottfried** [comment]
I guess basically it would be nice to have something like a stack trace

**Comment by maccallum** [comment]
i'm not sure that would do it either due to the kinds of transformations that happen to the code, but we could try
